crates/store/re_types_core/src/component_descriptor.rs

```diff
@@ -0,0 +1,177 @@
+use std::borrow::Cow;
+
+use crate::{ArchetypeFieldName, ArchetypeName, ComponentName, SizeBytes};
+
+/// A [`ComponentDescriptor`] fully describes the semantics of a column of data.
+///
+/// Every component is uniquely identified by its [`ComponentDescriptor`].
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
+pub struct ComponentDescriptor {
+    /// Optional name of the `Archetype` associated with this data.
+    ///
+    /// `None` if the data wasn't logged through an archetype.
+    ///
+    /// Example: `rerun.archetypes.Points3D`.
+    pub archetype_name: Option<ArchetypeName>,
+
+    /// Optional name of the field within `Archetype` associated with this data.
+    ///
+    /// `None` if the data wasn't logged through an archetype.
+    ///
+    /// Example: `positions`.
+    pub archetype_field_name: Option<ArchetypeFieldName>,
+
+    /// Semantic name associated with this data.
+    ///
+    /// This is fully implied by `archetype_name` and `archetype_field`, but
+    /// included for semantic convenience.
+    ///
+    /// Example: `rerun.components.Position3D`.
+    pub component_name: ComponentName,
+}
+
+impl std::hash::Hash for ComponentDescriptor {
+    #[inline]
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        let Self {
+            archetype_name,
+            archetype_field_name,
+            component_name,
+        } = self;
+
+        let archetype_name = archetype_name.map_or(0, |v| v.hash());
+        let archetype_field_name = archetype_field_name.map_or(0, |v| v.hash());
+        let component_name = component_name.hash();
+
+        state.write_u64(archetype_name ^ archetype_field_name ^ component_name);
```

Comment:
Surely that makes sense, right?

Reply:
why not do several `write_u64`? Should have roughly the same effect, but then this is owned by the hasher - afraid of perf penalty?

Reply:
We want this to be `NoHash`, so we're not allowed more than one write. I'll add a comment.